src/main/groovy/wooga/gradle/build/unity/tasks/AbstractUnityBuildEngineTask.groovy

```diff
@@ -0,0 +1,131 @@
+package wooga.gradle.build.unity.tasks
+
+
+import org.gradle.api.file.RegularFile
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import wooga.gradle.build.unity.internal.BuildEngineArgs
+import wooga.gradle.secrets.internal.Secrets
+import wooga.gradle.unity.UnityTask
+
+import javax.crypto.spec.SecretKeySpec
+
+
+class AbstractUnityBuildEngineTask extends UnityTask {

```

Comment:
It is strange to call the class abstract but the class is actually not abstract.

Reply:
I agree, as the UnityTask itself it not abstract either. I see prefixing Abstract to a class as a last resort for naming a root level class (something I occasionally do).
It is not something an user would expect after a concrete class.

Reply:
It is supposed to be abstract 😅

Reply:
@Azurelol this is the root (for this project, at least). I understand your concerns about the naming, but I really couldn't think on a better name for this